### ISSUE TITLE
Continuous training of a Ultralytics Model

### ISSUE THREAD

**Issue by ishanInstaCash** [ultralytics/yolov5#13242]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I have .pt file of my model for Object Detection it has 19 classes now I have some new data corresponding to only one class - class number 3 . I have created the YAML file in the format that ultralytics YOLO demand . Can someone guide me how do i train the model for this new data without losing previous information . 

I have tried resume but it gives me an error saying there is nothing to resume

**Comment by glenn-jocher** [comment]
@ishanInstaCash hello,

Thank you for your question! To continue training your existing YOLOv5 model with new data while retaining the knowledge from the previous training, you can follow these steps:

1. **Ensure Compatibility**: Make sure your new dataset is compatible with the original dataset in terms of class definitions. Your `dataset.yaml` file should include all 19 classes, even if your new data only contains instances of class 3.

2. **Prepare Your Dataset**: Organize your new data in the same structure as your original dataset. Ensure that the new images and labels are correctly formatted and placed in the appropriate directories.

3. **Update YAML File**: Your `dataset.yaml` should look something like this, ensuring it points to the new data paths:

    ```yaml
    path: ../datasets/your_dataset  # dataset root dir
    train: images/train  # train images (relative to 'path')
    val: images/val  # val images (relative to 'path')
    test:  # test images (optional)

    # Classes
    names:
        0: class0
        1: class1
        2: class2
        3: class3
        ...
        18: class18
    ```

4. **Train with Pretrained Weights**: Use your existing `.pt` file as pretrained weights. This will allow the model to retain the knowledge from the previous training while learning from the new data. You can do this by specifying the path to your `.pt` file in the `--weights` argument:

    ```bash
    python train.py --img 640 --epochs 10 --data dataset.yaml --weights path/to/your_model.pt
    ```

5. **Monitor Training**: Keep an eye on the training process to ensure that the model is learning effectively from the new data without overfitting.

If you encounter any issues, please ensure you are using the latest version of YOLOv5 by pulling the latest changes from the repository and updating your dependencies:

```bash
git pull
pip install -r requirements.txt
```

This approach should help you continue training your model with the new data while retaining the previous knowledge. If you have any further questions or run into any issues, feel free to ask!

**Comment by ishanInstaCash** [comment]
Thank you so much for the help

**Comment by glenn-jocher** [comment]
@ishanInstaCash you're welcome! 😊 I'm glad I could assist you. If you have any more questions or run into any issues, feel free to reach out. The YOLOv5 community and the Ultralytics team are always here to help.

Happy training and best of luck with your object detection project! 🚀

If you encounter any bugs or have further questions, please ensure you're using the latest version of YOLOv5 and check the [documentation](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) for additional resources.

Have a great day!
